### PR TITLE
kavita: 0.7.1.4 -> 0.7.11.2

### PR DESCRIPTION
**Pull request by nevivurn**
## Description of changes

Supersedes #242739

- Add myself as maintainer.
- Add updateScript
- Update 0.7.1.4 -> 0.7.11.2 ([changes](https://github.com/Kareadita/Kavita/compare/v0.7.1.4...v0.7.11.2)).
  - Update patches to apply cleanly, also add patches & tests to find locale files.
- Update nixos module to accept freeform settings.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/web-apps/kavita.nix
Comment (Misterio77) [inline nixos/modules/services/web-apps/kavita.nix:35]: Damn that was a typo on my part, my bad :skull:

**Review by Misterio77 — CHANGES_REQUESTED** [review]
LGTM, just gotta add the mkRenamedOptionModule's before merging

File: nixos/modules/services/web-apps/kavita.nix
Comment (nevivurn) [reply nixos/modules/services/web-apps/kavita.nix:8]: I used `mkChangedOptionModule` for ipAddresses, since the type was changed from a list of strings to a comma-separated string.

Comment (Misterio77) [reply nixos/modules/services/web-apps/kavita.nix:8]: Nice!

**Review by Misterio77 — APPROVED** [review]
LGTM

**Comment by nevivurn** [comment]
Since last review:
- Updated to latest (0.7.10.2 -> 0.7.11.2)
- Added myself as maintainer
- Added updateScript

**Comment by nevivurn** [comment]
superseded by #281236

**Pull request closed without merging**
